src/util/auth.cpp

```diff
@@ -135,3 +142,21 @@ bool decode_srp_verifier_and_salt(const std::string &encoded,
 	return true;
 
 }
+
+/// Override every character before clearing
+void clear_string(std::string &text)
+{
+	#ifdef __STDC_LIB_EXT1__
+	memset_s((void *)text.data(), text.size(), '0', text.size());
+	#elif _WIN32
+	SecureZeroMemory((void *)text.data(), text.size());
+	#else
+	volatile char *ch = (char *)text.data();
+	size_t n = text.size();
+	for (;n>0;n--) {
+		*ch = 0;
+		ch++;
+	}
+	#endif
+	text.clear();
+}
```

Comment:
this looks like it should be in porting.cpp

I would also design the interface as follows:
```cpp
void ZeroMemory(void *ptr, size_t count);
inline void ZeroMemory(std::string &s) {
    ZeroMemory(&s[0], s.size());
}
```

Reply:
Only moved now.